### DISCUSSION TITLE
Questions about the language/compiler

### DISCUSSION THREAD

**Discussion by Its-Jakey** [Its-Jakey/SCPP#1] [Q&A]
Have a question? Ask it here!

**Comment by MinekPo1** [comment]
Is the type system staying as is, or will it change?

**Reply by Its-Jakey** [reply]
The type system is staying

**Comment by MinekPo1** [comment]
Can multidimensional arrays have differently sized sub arrays?

Like:

```SCPP
var arr = {{1,2}, {1,2,3}
```

**Reply by Its-Jakey** [reply]
Yes, arrays dimensions can be differently sized. It just gets hard to keep track of the sizes

**Comment by MinekPo1** [comment]
Do for and while loops act like namespaces?

like:

```scpp
var n = 0;
while(n != 2){
    var a = 0;
    n++;
}
print(a);
```

Does this cause a name error when printing?

**Reply by Its-Jakey** [reply]
No. When variables are defined in functions and not at top level, the variable will be accessible at the top level of the function body
